### ISSUE TITLE
bug in a simple series expansion

### ISSUE THREAD

**Issue by fchapoton** [sagemath/sage#38868]
### Steps To Reproduce

```
sage: var('t')
sage: v = -125/24*e^(-5*t)/t - 8/3*e^(-4*t)/t^2 - 3/2*e^(-3*t)/t^3 - e^(-2*t)/t^4 - e^(-t)/t^5 + 1/t^5; v
-125/24*e^(-5*t)/t - 8/3*e^(-4*t)/t^2 - 3/2*e^(-3*t)/t^3 - e^(-2*t)/t^4 - e^(-t)/t^5 + 1/t^5
sage: v.series(t,2)
54/5 + (-3731/90)*t + Order(t^2)
sage: v.series(t,4)
54/5 + (-29849/720)*t + 139459/1680*t^2 + (-2305729/20160)*t^3 + Order(t^4)
sage: v.series(t,6)
54/5 + (-29849/720)*t + 139459/1680*t^2 + (-1537153/13440)*t^3 + 43947619/362880*t^4 + (-10845131/103680)*t^5 + Order(t^6)

```

### Expected Behavior

answer should be correct up to the required order.

### Actual Behavior

last term is always wrong

### Environment

- **OS**: Ubuntu 24.04
- **Sage Version**: 10.5.beta8

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

**Comment by mantepse** [comment]
what's `u`?

**Comment by fchapoton** [comment]
if needed, I can provide many similar examples. The smallest is

`-e^(-5*t)/t - 4*e^(-t)/t^5 + 6*e^(-2*t)/t^5 - 4*e^(-3*t)/t^5 + e^(-4*t)/t^5 + 1/t^5`

**Comment by fchapoton** [comment]
a few more data:
```

sage: g=-e^(-2*t)/t - e^(-t)/t^5
sage: bool(g.series(t,2)==g.series(t,4).series(t,2))
False
sage: g=-e^(-2*t)/t - e^(-t)/t^4
sage: bool(g.series(t,2)==g.series(t,4).series(t,2))
True

```

**Comment by maxale** [comment]
Is it related by any chance to the bug in `taylor()` reported at https://ask.sagemath.org/question/79842 ?

**Comment by fchapoton** [comment]
I do not think so. The method `series` is handled by `pynac/ginac`, but `taylor` is done by `maxima`.

**Comment by fchapoton** [comment]
> Worse:
> 
> ```
> sage: g = e^(t) - e^(t)/t^6
> sage: g.series(t, 1)
> (-1)*t^(-6) + (-1)*t^(-5) + (-1/2)*t^(-4) + (-1/6)*t^(-3) + 1 + Order(t)
> sage: g.series(t, 2)
> (-1)*t^(-6) + (-1)*t^(-5) + (-1/2)*t^(-4) + (-1/6)*t^(-3) + (-1/24)*t^(-2) + 1 + 1*t + Order(t^2)
> ```

This also works fine in `GiNaC V1.8.7`

**Comment by fchapoton** [comment]
many fixes have been made to the series code in ginac, see

https://www.ginac.de/ginac.git/?p=ginac.git&a=search&h=HEAD&st=commit&s=series

Our ginac code base is an old and very obsolete version.
